packages/server/src/index.ts

```diff
@@ -136,35 +138,27 @@ function createServerAdapter<
     if (waitUntilPromises.length > 0) {
       await handleWaitUntils(waitUntilPromises);
     }
-    return response;
   }
 
-  function handleEvent(event: FetchEvent, ...ctx: Partial<TServerContext>[]) {
+  function handleEvent(event: FetchEvent, ...ctx: Partial<TServerContext>[]): void {
     if (!event.respondWith || !event.request) {
       throw new TypeError(`Expected FetchEvent, got ${event}`);
     }
-    let serverContext = {} as TServerContext;
-    if (ctx?.length > 0) {
-      serverContext = Object.assign({}, serverContext, ...ctx);
-    }
+    const serverContext = Object.assign({}, event, ...ctx);
```

Comment:
`FetchEvent` should be part of the context to access `waitUntil` etc.

Reply:
Ah ok, I understand.